util/cpb/main.go

```diff
@@ -128,3 +161,107 @@ func getMetadata() (m *metadata, err error) {
 
 	return
 }
+
+// copyIndex searches for the index in an image and then copies it to dest
+// it will create intermediate directories for dest if they do not exist
+func copyIndex(dest string) (err error) {
+	path := getIndexPath()
+	if path == "" {
+		err = fmt.Errorf("could not find index data in image")
+		return
+	}
+
+	dir := filepath.Dir(dest)
+	if err = os.MkdirAll(dir, os.ModePerm); err != nil {
+		return
+	}
+
+	in, err := os.Open(path)
+	if err != nil {
+		return
+	}
+	defer in.Close()
+	out, err := os.Create(dest)
+	if err != nil {
+		return
+	}
+	defer func() {
+		cerr := out.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
+
+	hash := sha256.New()
+	hashAndCopy := io.MultiWriter(hash, out)
+
+	if _, err = io.Copy(hashAndCopy, in); err != nil {
+		return
+	}
+	if err = out.Sync(); err != nil {
+		return
+	}
+
+	status := reconciler.CopyIndexResult{Digest: fmt.Sprintf("%x", hash.Sum(nil))}
+	b, err := json.Marshal(status)
+	if err != nil {
+		return
+	}
+	return ioutil.WriteFile(reconciler.CopyIndexStatusLocation, b, os.ModePerm)
+}
+
+// getIndexPath searches standard paths for a db file and then falls back to walk the fs.
+// it is possible to get a false positive if there is more than one sqlite file in the image,
```

Comment:
does the bundle have an annotation that provides the extract db location? or just its name. 

I guess as you said it doesn't matter, in reality only one sqlite db will be present in the index at any time. Intuitively that makes sense but I don't recall if we have index-level validation that enforces that rule.